File: 1.two-sum.ts
/*
 * @lc app=leetcode id=1 lang=typescript
 *
 * [1] Two Sum
 */

// @lc code=start
function twoSum(nums: number[], target: number): number[] {
  // Declaration of a new map constant to store the calculations
  const map = new Map<number, { index: number }>();

  // Iteration over the array nums
  for(let i = 0; i < nums.length; i++) {
    const num = target - nums[i];
    const part = map.get(num);

    if (part) return [part.index, i];

    map.set(nums[i], { index: i});
  }

  return [];
};
// @lc code=end

File: 1.two-sum.c
/*
 * @lc app=leetcode id=1 lang=c
 *
 * [1] Two Sum
 */

// @lc code=start
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize){
    *returnSize=2;// der zu returnende array hat 2 elemente

    int *array = malloc(2*sizeof(int) ); // Output array max 2 elemente

    for (int i=0; i<numsSize;i++)
    {
        for( int j=1+i;j<numsSize;j++)
        {
            if(nums[i] + nums[j] == target)
            {
            array[0]=i;
            array[1]=j;
            return array;
            }
        }   
         
}
return -1;

}
// @lc code=end
